lib/plugins/aws/package/index.js

```diff
@@ -52,7 +52,11 @@ class AwsPackage {
        */
       'package:cleanup': () =>
         BbPromise.bind(this)
-          .then(() => this.serverless.pluginManager.spawn('aws:common:validate'))
+          .then(() => {
+            if (this.serverless.processedInput.commands[0] === 'package') {
```

Comment:
Why exactly do we need this condition?

Technically validation is already skipped for not applicable commands here: https://github.com/serverless/serverless/blob/f767fab43952c5a394d76b6b0f3f9779c8a5a3e6/lib/plugins/aws/lib/validate.js#L19-L24

Reply:
True that validation is skipped there for non-applicable commands..but for commands like `sls deploy` this invokes it **twice**. Running with `SLS_DEBUG` shows this. So adding this check removes the need for an initial check during the _packaging_ phase whilst preserving the existing validation check if a user wants to explicitly run the packaging command: `sls package`

Reply:
Ok, I think it's logical error, that `aws:common:validate` is called here at all.

It's purely about validation of an AWS credentials, and this validation is already not pursued with `package` command (due to conditional I've linked above), while in `deploy` it incurs a double validation.

Therefore I think it's totally safe to remove this instruction completely here